src/components/d3/TimeXAxisGroup.js

```diff
@@ -0,0 +1,54 @@
+const React = require('react');
+
+const d3 = require('d3');
+const moment = require('moment');
+
+const TimeXAxisGroup = React.createClass({
+  propTypes: {
+    numberOfTicks: React.PropTypes.number,
+    tickSize: React.PropTypes.number,
+    timeAxisFormat: React.PropTypes.string.isRequired,
+    translation: React.PropTypes.string,
+    xScaleFunction: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      numberOfTicks: 10,
+      tickSize: 6,
+      translation: 'translate(0,0)'
+    };
+  },
+
+  componentDidMount () {
+    this._renderAxis();
+  },
+
+  componentDidUpdate () {
+    this._renderAxis();
+  },
+
+  _renderAxis () {
+    const timeAxisFunction = d3.svg.axis()
+    .scale(this.props.xScaleFunction())
+    .tickFormat(d => {
+      return moment.unix(d).format(this.props.timeAxisFormat);
+    })
+    .tickSize(this.props.tickSize, this.props.tickSize)
```

Comment:
Yes, the D3 tickSize function takes two arguments.  Inner and Outer tick sizes.  I just want them to be the same size in this case.